modules/sor/sorV2/lib/utils/helpers.ts

```diff
@@ -77,41 +79,53 @@ export function getOutputAmount(paths: PathWithAmount[]): TokenAmount {
     return amounts.reduce((a, b) => a.add(b));
 }
 
-export function getHookState(pool: any): HookState | undefined {
-    if (pool.hook === undefined || pool.hook === null) {
+export function getHookState(pool: PrismaPoolAndHookWithDynamic): HookState | undefined {
+    if (!pool.hook) {
         return undefined;
     }
 
-    if (pool.hook.name === 'ExitFee') {
-        // api for this hook is an Object with removeLiquidityFeePercentage key & fee as string
-        const dynamicData = pool.hook.dynamicData as { removeLiquidityFeePercentage: string };
-
-        return {
-            tokens: pool.tokens.map((token: { address: string }) => token.address),
-            // ExitFeeHook will always have dynamicData as part of the API response
-            removeLiquidityHookFeePercentage: parseEther(dynamicData.removeLiquidityFeePercentage),
-            hookType: pool.hook.name,
-        };
-    }
-
-    if (pool.hook.name === 'DirectionalFee') {
-        // this hook does not require a hook state to be passed
-        return {
-            hookType: pool.hook.name,
-        } as HookState;
+    const hookData = pool.hook as HookData;
+
+    switch (hookData.type) {
+        case 'EXIT_FEE': {
+            // api for this hook is an Object with removeLiquidityFeePercentage key & fee as string
+            const dynamicData = hookData.dynamicData as { removeLiquidityFeePercentage: string };
+
+            return {
+                tokens: pool.tokens.map((token: { address: string }) => token.address),
+                // ExitFeeHook will always have dynamicData as part of the API response
+                removeLiquidityHookFeePercentage: parseEther(dynamicData.removeLiquidityFeePercentage),
+                hookType: 'ExitFee',
```

Comment:
I thought about hard coding the hookType back then as well. What would be the advantage of hardcoding it here compared to forwarding the `hookData.name`?

Reply:
Just FYI, hook.name is being deprecated in the favor of hookType

Reply:
do we need this new hookType that is returned here for the Balancer maths repo?

Reply:
Answering @mkflow27 - the advantage of rewriting the hook type here instead of forwarding is that we decouple how they are declared in the api vs in Balancer Maths. If we simply forwarded what was specified in the api, we add the overhead of having to sync and reach consensus on how each hook type is handled in both codebases.
Since we always need to add code to support new hook types on the SOR, adding another attribute is trivial.